### ISSUE TITLE
Update API to support user defined key-value metadata for variables

### ISSUE THREAD

**Issue by kyao** [usc-isi-i2/datamart-api#60]
Support user defined key-value metadata for variables.  Add an additional field to variable metadata called `user_defined` and the content of this field is a dictionary containing key-value pairs.

```
{
    'name': 'a variable', 
    'description': 'description of a variable',
    'dataset_id': 'a_dataset_id',
    'user_defined': {
        'FactorClass': 'EconomicAgriculturalCapability',
        'Relevance': 1,
        'Normalizer': 'Longitudinal'
    }
}
```

Augment API query to support querying user define key value pairs.

The following returns all variables with user defined key `FactorClass` that as value `EconomicAgriculturalCapability`:
```
`/metadata/variables?FactorClass=EconomicAgriculturalCapability'
```

The following returns all variables with user defined key `FactorClass:
```
`/metadata/variables?exists=FactorClass'
```

**Comment by kyao** [comment]
Augment data API: return tags as columns

```
GET /datasets/a_dataset_id/variables/a_variable_id
```

```
FactorClass, Relevance,Normalizer
EconomicAgriculturalCapability, 1, Longitudinal
EconomicAgriculturalCapability, 1.5, Simple
Transportation, 1, Simple
```

**Comment by zmbq** [comment]
I'm not sure how the tags are supposed to be returned as columns - what is the value of that column for each row? Since the tags are for the entire variable, the columns will all contain the same value. We also treat tags as a whole, so the same variable can have Relevance:1 and Relevance:5 as tags.

**Comment by zmbq** [comment]
Add a column for each tag A (A:B), the value should be B. If there are A:B and A:C tags, return B|C as the cell value

**Comment by zmbq** [comment]
Columns are added for all tags automatically. You can remove specific tag columns by specifying `?exclude=tag-name` in the query.

**Comment by saggu** [comment]
From the spreadsheet, they do not want the following columns as tags:
-  File Type
-  Time Series Data File
-  Indicator Code
-  Indicator Name

They **DO** need the following:
- doc id


Factor class can also have an IRI instead of just name

**Comment by szeke** [comment]
So, this means we don't create tags for these columns, correct? Otherwise we need to invent a way to specify which tags become columns.

**Comment by saggu** [comment]
Yes, we will not create tags for these columns. All tags will be returned as columns in the GET data API

**Comment by saggu** [comment]
database backup containing the tags - https://drive.google.com/file/d/1buKO4UghTGOY_gVnPeW8rW-t4yKnSWiq/view?usp=sharing
